### PR TITLE
[#436] Add Facility model validation

### LINKED ISSUE #436: Add model level validations to Facility
Add model level validations to Facility

- [ ] name: presence
- [ ] code: presence and uniqueness (scoped to organization)

@CraigJZ Do these sound reasonable to you?

### PR DESCRIPTION
**Pull request by nsiregar**
Resolves #436  

### Description

Add model level validation to `Facility` which do
- validates the presence of the name
- validates the presence of code
- validates the uniqueness of code, scoped to organization_id
### Type of change

* New feature (non-breaking change which adds functionality)

### How Has This Been Tested?

Tested with rspec matcher but with some issue regarding shoulda-matchers issue here https://github.com/thoughtbot/shoulda-matchers/issues/814

Instead of using `organization` object, it is scoped to `organization_id`

### CODE REVIEW

File: spec/features/facilities/index_spec.rb [hunk 1]
```diff
@@ -29,7 +29,11 @@
   end
 
   it "should allow exporting to csv" do
-    facilities = create_list(:facility, 5, organization: user.organization)
+    facility_codes = %w[vortex anemo dendro pyro electro]
+    facilities = []
+    facility_codes.each do |code|
+      facilities << create(:facility, code: code, organization: user.organization)
+    end
```
Comment (benreyn) [inline spec/features/facilities/index_spec.rb:36]: Instead of this, could we update the [facility factory](https://github.com/rubyforgood/abalone/blob/master/spec/factories/facility.rb) to use a [sequence](https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#sequences)? The `create_list` approach should still work in that case.

File: spec/models/facility_spec.rb [hunk 2]
```diff
@@ -16,6 +16,11 @@
 RSpec.describe Facility, type: :model do
   let(:facility) { FactoryBot.create :facility }
 
+  it 'includes required modules' do
```
Comment (benreyn) [inline spec/models/facility_spec.rb:19]: These changes dont seem related to this PR. Would you mind making a separate PR for these changes that explains why these changes are needed?

**Review by benreyn — COMMENTED** [review]
Thanks for tackling this for us @nsiregar! I left a few notes, but Ill also wait for @CraigJZ to weigh in.

File: spec/features/facilities/index_spec.rb (hunk 1 above)
Comment (CraigJZ) [reply spec/features/facilities/index_spec.rb:36]: Yes, thanks for taking on this issue @nsiregar. I agree, if we could use the `create_list` approach here, that would be great.

Comment (nsiregar) [reply spec/features/facilities/index_spec.rb]: Sure, we can use factory sequence

File: spec/models/facility_spec.rb (hunk 2 above)
Comment (nsiregar) [reply spec/models/facility_spec.rb]: Will do it

**Review by benreyn — APPROVED** [review]
This looks good! Thanks so much @nsiregar!
